### PR TITLE
RUST-695 Retry `ConnectionPoolCleared` errors

### PR DESCRIPTION
**Pull request by patrickfreed**
RUST-695

This PR updates the operation execution logic to retry pool cleared errors once. It also updates the message contained in a pool cleared error to contain the originating error.

### CODE REVIEW

File: src/cmap/test/mod.rs [hunk 1]
```diff
@@ -285,7 +285,13 @@ impl Operation {
             }
             Operation::Clear => {
                 if let Some(pool) = state.pool.read().await.as_ref() {
-                    pool.clear().await;
+                    pool.clear(
```
Comment (abr-egn) [inline src/cmap/test/mod.rs:288]: Is this only executed by test code?

Comment (abr-egn) [reply src/cmap/test/mod.rs:288]: Derp, just saw the file path.

**Review by abr-egn — APPROVED** [review]

File: src/client/executor.rs [hunk 2]
```diff
@@ -129,6 +129,9 @@ impl Client {
 
         let mut conn = match server.pool.check_out().await {
             Ok(conn) => conn,
+            Err(err) if err.is_pool_cleared() => {
+                return self.execute_retry(&mut op, &mut session, None, err).await
```
[the remark this answers is not in the document]
Comment (patrickfreed) [reply src/client/executor.rs]: done. It made me curious whether we ought to add the transient transaction error label to this error. I asked shane about it, might have a forthcoming DRIVERS ticket to require it if so.

[the remark this answers is not in the document]
Comment (patrickfreed) [reply src/client/executor.rs]: It looks like we will need to, but I'll hold off on that until DRIVERS-1815 is specified before attaching the label here.

**Review by isabelatkinson — APPROVED** [review]

**Pull request merged**